src/lib/components/Channel/Chat/DrawerRestream.svelte

```diff
@@ -199,10 +215,30 @@
 						<div class="form-control w-full pt-4">
 							<div class="flex gap-3">
 								{#if env.PUBLIC_CROSS_ORIGIN === 'false'}
-									<a class="btn btn-sm" href="{env.PUBLIC_API_URL}/auth/twitch"
-										><IconSocialTwitch /> Twitch</a>
-									<a class="btn btn-sm" href="{env.PUBLIC_API_URL}/auth/youtube"
-										><IconSocialYouTube /> YouTube</a>
+									<button
+										on:click={async () => {
+											const linkRes = await get(
+												`youtube/link?channelId=${$page.params.channelId}`,
+												auth
+											)
+											if (linkRes.redirect) {
+												window.location.replace(linkRes.redirectUrl)
+											}
+										}}>Link Youtube</button>
+									<button
+										on:click={async () => {
+											const linkRes = await get(
+												`twitch/link?channelId=${$page.params.channelId}`,
+												auth
+											)
+											if (linkRes.redirect) {
+												window.location.replace(linkRes.redirectUrl)
+											}
+										}}>Link twitch</button>
+									<button class="btn btn-sm" on:click={getTwitchStreamLink}
+										><IconSocialTwitch /> Twitch</button>
+									<button class="btn btn-sm" on:click={getYoutubeStreamLink}
+										><IconSocialYouTube /> YouTube</button>
```

Comment:
Theres 4 buttons here.
1. There should be 2 buttons in the if (for prod login) and 2 in the else (for dev login).
2. Login should be handled in hooks.server, not with the windows.location.replace. see comments in hooks.server.